isthmus/src/test/java/io/substrait/isthmus/SubstraitRelNodeConverterTest.java

```diff
@@ -131,6 +131,17 @@ public void emit() {
       var relNode = converter.convert(root.getInput());
       assertRowMatch(relNode.getRowType(), R.I32, N.STRING);
     }
+
+    @Test
+    public void singleOrList() {
+      Plan.Root root =
+          b.root(
+              b.filter(
+                  input -> b.singleOrList(b.fieldReference(input, 0), b.i32(5), b.i32(10)),
+                  commonTable));
+      var relNode = converter.convert(root.getInput());
```

Comment:
The plan here does get created with a SEARCH
```
LogicalFilter(condition=[SEARCH($0, Sarg[5, 10])])
  LogicalTableScan(table=[[example]])
```

So it does test the SEARCH path, but only from Substrait -> Calcite. We should add a check that for going from Calcite -> Substrait. 

Also, I'm not sure if this is the best place for this test. All the other tests in the file effectively check the record layout of various relations, whereas we're interested verifying the conversion of the SingleOrList _expression_ which won't affect the record layout and isn't tied to a specific relation.

I would advocate for having something like `ExpressionRoundtripTest` and adding a roundtrip test inside of that. There are some other expressions that get created differently depending on whether you use SQL as input or Substrait as input, so I imagine we will add more to a test file like that.

Reply:
`IN` statements won't roundtrip cleanly when dealing with Calcite. A `SEARCH` expression gets translated into a series of `OR` statements using `RexUtil.expandSearch`, which is how Isthmus tries to [translate `SEARCH`](https://github.com/substrait-io/substrait-java/blob/main/isthmus/src/main/java/io/substrait/isthmus/expression/CallConverters.java#L79-L89).

I added a test that does ensure that we still generate valid Substrait from this plan, but I'm not sure `ExpressionRoundtripTest` is the right name. The test is really for similar (and semantically equivalent) queries, but not necessarily of identical expressions.

Reply:
> IN statements won't roundtrip cleanly when dealing with Calcite.
> ...
> I'm not sure ExpressionRoundtripTest is the right name.

Good point. I'm open to a different/better name if you have one.

Reply:
I've renamed to ExpressionConvertabilityTest, since we're testing the ability to convert to and from Calcite here.